### ISSUE TITLE
nd.contrib.fft issue

### ISSUE THREAD

**Issue by DabiaoMa** [apache/incubator-mxnet#9302]
Hi,

I am currently using nd.contrib.fft to compute STFT(short time Fourier transform) of audio wavs to implement the distilled WaveNet. 
I noticed that even I set the weight of STFT loss to 0 (0 * STFT_loss), there are still gradients backward from STFT_loss to parameters, and this behavior causes the gradient to be NaN if the weight is not 0 (1 * STFT_loss)
I have triple checked my implemented and this is the only possible place that may raise an error.

Best,

**Comment by shiyangdaisy23** [comment]
Hi,

Can you provide some reproducible example?

Thanks

**Comment by DabiaoMa** [comment]
@shiyangdaisy23 
Here is an simple example:
######################

from mxnet import nd, gpu, autograd, gluon
import numpy as np
import librosa

ctx = gpu(2)
fft_length = 256
fft_size = fft_length // 2 + 1

f = gluon.nn.Dense(1, flatten=False)
f.collect_params().initialize(ctx=ctx)

wavs = nd.array(librosa.load('210001.wav', sr=16000, mono=True)[0], ctx=ctx)
x = nd.random.uniform(shape=(1, wavs.shape[0], 1), ctx=ctx)

third_dim = x.shape[1] // fft_length + 1
padding_needed = wavs.shape[0] % fft_length

wavs = nd.expand_dims(nd.expand_dims(wavs, axis=0), axis=0)

f(x)

print f.weight.grad()

with autograd.record():
    y = f(x)
    y = nd.swapaxes(y, dim1=1, dim2=2)
    y = nd.concat(y, nd.zeros(shape=(1, 1, padding_needed), ctx=ctx), dim=2).reshape((1, 1, -1, fft_length))
    wavs = nd.concat(wavs, nd.zeros(shape=(1, 1, padding_needed), ctx=ctx), dim=2).reshape((1, 1, -1, fft_length))
    stft_y = nd.contrib.fft(y, compute_size=fft_length).reshape((1, 1, third_dim, -1, 2))
    stft_wavs = nd.contrib.fft(y, compute_size=fft_length).reshape((1, 1, third_dim, -1, 2))
    stft_y = nd.sqrt(nd.sum(nd.square(stft_y), axis=4) + 1e-12)[:, :, :, : fft_size]
    stft_wavs = nd.sqrt(nd.sum(nd.square(stft_wavs), axis=4) + 1e-12)[:, :, :, : fft_size]
    loss = 0 * nd.mean(nd.square(stft_y - stft_wavs), axis=(1, 2, 3))

loss.backward()
print f.weight.grad()
####################

The version I am using is 0.12.0, with support of cu8.0
Initially the grad of f.weight is 0. After the backward operation, the grad is supposed to be 0( because the loss is multiplied by 0), but I got 81.09 instead

**Comment by DabiaoMa** [comment]
@shiyangdaisy23 
Sorry, it is my mistake, the number of padding zeros are not correct. There is nothing wrong with nd.contrib.fft

**Comment by anirudhacharya** [comment]
@sandeep-krishnamurthy please close this issue. it is resolved.
